src/request-pipeline/connection-reset-guard.js

```diff
@@ -5,8 +5,10 @@ const connectionResetDomain = domain.create();
 
 connectionResetDomain.on('error', err => {
     // NOTE: The nodejs throw the EPIPE error instead of the ECONNRESET error
-    // when the connection is broken in some cases on MacOS
-    if (err.code === 'ECONNRESET' || os.type() === 'Darwin' && err.code === 'EPIPE')
+    // when the connection is broken in some cases on MacOS and Linux
+    const isMacOrLinux = os.type() === 'Darwin' || os.type() === 'Linux';
```

Comment:
Let's use the [os-family](https://www.npmjs.com/package/os-family) module to simplify readability.